include/deal.II/lac/vector.templates.h

```diff
@@ -986,11 +1023,14 @@ Vector<Number>::memory_consumption () const
 
 template <typename Number>
 void
-Vector<Number>::allocate()
+Vector<Number>::allocate(const size_type copy_n_el)
 {
   // allocate memory with the proper alignment requirements of 64 bytes
   Number *new_values;
   Utilities::System::posix_memalign ((void **)&new_values, 64, sizeof(Number)*max_vec_size);
+  // copy:
+  for (size_type i = 0; i < copy_n_el; ++i)
+    new_values[i] = values[i];
```

Comment:
Should we use the old TBB partitioner and some of the internal vector operations to do this?

Reply:
i was concerned with this operation for dense linear algebra where vector size is `[1,20)` or so, so in this case it's not crucial. I would prefer to leave it for further PRs if/when someone applies it to larger vectors.

Reply:
Yes sure.